### PR TITLE
fix: use non-root user

### PR DESCRIPTION
**Pull request by matejvasek**

### CODE REVIEW

File: buildpacks/go/bin/build [hunk 1]
```diff
@@ -24,7 +24,7 @@ EOF
 cp $build_dir/bin/faas "$app_layer/faas"
 
 # REMOVE SOURCE CODE FROM FINAL CONTAINER
-rm -rf $build_dir/*
+# rm -rf $build_dir/*
```
Comment (matejvasek) [inline buildpacks/go/bin/build:27]: @lkingland is this useful? is it really in run image?

**Comment by matejvasek** [comment]
According to spec user should be set to non root user. Otherwise with newer `pack` there are errors: `ERROR: failed to build: refusing to run as root`.

**Review by lance — APPROVED** [review]
I think I may have introduced this regression. Thanks for catching it, @matejvasek

File: buildpacks/go/bin/build (hunk 1 above)
Comment (lkingland) [reply buildpacks/go/bin/build:27]: I don't think this line is necessary.

This line was added on request from someone else (been a long time now; cant remember who). I was  pretty sure, when I put it together originally, that source was not included in the final run image.  But right now I can not remember exactly how the layers get assembled.  Sorry I can't remember more ATM!

**Review by lkingland — APPROVED** [review]

**Comment by matejvasek** [comment]
@lance BTW I think that with this we might disable `--trust-builder` in `func` making it possible to use `podman` (before we fix root-cause  `podman`)

File: buildpacks/go/bin/build [hunk 2]
```diff
@@ -23,9 +23,6 @@ EOF
 
 cp $build_dir/bin/faas "$app_layer/faas"
 
-# REMOVE SOURCE CODE FROM FINAL CONTAINER
-rm -rf $build_dir/*
```
Comment (lance) [inline buildpacks/go/bin/build:27]: @matejvasek I actually think this is a good thing to do. It keeps the source out of the runtime image. And more importantly, it removes func.yaml (which may contain sensitive data) from the run image.

Comment (matejvasek) [reply buildpacks/go/bin/build:27]: @lance there was an issue with deleting it while not being a root: stuff in `/workspace/go/pkg/mod/` seems to have `-r--r--r--` rights

Comment (matejvasek) [reply buildpacks/go/bin/build:27]: also do we remove those files in other BPs?

**Pull request merged**